### PR TITLE
Updates CLI to use '@octokit/rest' instead of 'github'.

### PR DESCRIPTION
**Pull request by bicknellr**

### CODE REVIEW

File: packages/cli/src/github/github.ts [hunk 1]
```diff
@@ -156,21 +154,22 @@ export class Github {
...
-    const releases: GitHubApi.Release[] = await this._github.repos.getReleases({
+    const response: Octokit.AnyResponse = await this._github.repos.getReleases({
       owner: this._owner,
       repo: this._repo,
       per_page: 100,
     });
+    const releases = response.data;
     const validReleaseVersions =
-        releases.filter((r) => semver.valid(r.tag_name)).map((r) => r.tag_name);
+        releases.filter((r: any) => semver.valid(r.tag_name)).map((r: any) => r.tag_name);
```
Comment (bicknellr) [inline packages/cli/src/github/github.ts]: Unfortunately, it seems like their `index.d.ts` no longer has types for responses.

**Comment by rictic** [comment]
What's the advantage of `octokit` over `github`?

**Comment by TimvdLippe** [comment]
There is no advantage other than that the old package name is no longer updated. They simply renamed their package and continued development under the new name.

**Comment by bicknellr** [comment]
I was trying to figure out how to retrieve a branch instead of a release and noticed they had updated and couldn't find documentation for the version we were using.

**Comment by TimvdLippe** [comment]
See also https://github.com/octokit/rest.js/releases/tag/v14.0.0 btw

File: packages/cli/src/github/github.ts [hunk 2]
```diff
@@ -156,21 +154,29 @@ export class Github {
    * Get all Github releases and match their tag names against the given semver
    * range. Return the release with the latest possible match.
    */
-  async getSemverRelease(semverRange: string): Promise<GitHubApi.Release> {
+  async getSemverRelease(semverRange: string): Promise<any> {
```
Comment (rictic) [inline packages/cli/src/github/github.ts]: Switching from types to `any` is a pretty big regression. I'd rather not update the library if that's the primary change.

Comment (rictic) [reply packages/cli/src/github/github.ts]: If there's an expectation that the newer version of the library is better then I'm on board

Comment (TimvdLippe) [reply packages/cli/src/github/github.ts]: I think this is blocked on https://github.com/octokit/rest.js/pull/732

**Pull request closed without merging**